web-app/js/portal/cart/PythonDownloadHandler.js

```diff
@@ -37,7 +37,7 @@ Portal.cart.PythonDownloadHandler = Ext.extend(Portal.cart.DownloadHandler, {
         var _this = this;
 
         return function(collection) {
-            return collection.getLayerState().getSelectedLayer().getFeatureRequestUrl(
+            return OpenLayers.Layer.WMS.prototype.getFeatureRequestUrl(
```

Comment:
Given that we're not calling this function on a WMS layer it is essentially a static method. So we could have a util class to house it. WmsRequestUtils, or something. At least then we wouldn't be calling it on an object's prototype.